dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/acl/TrackedEntityInstanceAclReadTests.java

```diff
@@ -0,0 +1,354 @@
+package org.hisp.dhis.tracker.acl;
+
+import com.google.common.collect.Lists;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.apache.commons.collections.ListUtils;
+import org.hamcrest.Matcher;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.hamcrest.beans.HasProperty;
+import org.hamcrest.collection.IsEmptyCollection;
+import org.hamcrest.core.IsNot;
+import org.hisp.dhis.ApiTest;
+import org.hisp.dhis.actions.LoginActions;
+import org.hisp.dhis.actions.RestApiActions;
+import org.hisp.dhis.actions.UserActions;
+import org.hisp.dhis.actions.metadata.MetadataActions;
+import org.hisp.dhis.actions.tracker.TEIActions;
+import org.hisp.dhis.dto.ApiResponse;
+import org.hisp.dhis.dto.Me;
+import org.hisp.dhis.dto.OrgUnit;
+import org.hisp.dhis.dto.UserGroup;
+import org.hisp.dhis.helpers.QueryParamsBuilder;
+import org.hisp.dhis.helpers.models.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Stian Sandvold
+ */
+public class TrackedEntityInstanceAclReadTests
+    extends ApiTest
+{
+    private static final String _DATAREAD = "..r.*";
+
+    private MetadataActions metadataActions;
+
+    private UserActions userActions;
+
+    private TEIActions teiActions;
+
+    private static final List<User> users = new ArrayList<>();
+
+    @BeforeAll
+    public void before()
+        throws Exception
+    {
+        teiActions = new TEIActions();
+        metadataActions = new MetadataActions();
+        userActions = new UserActions();
+
+        // Setup as SuperUser
+        new LoginActions().loginAsDefaultUser();
```

Comment:
that is admin: district, not necessarily super user. If you want super, you can use loginAsSuperuser()